python-client/giskard/slicing/slice.py

```diff
@@ -11,59 +13,67 @@
 from ..ml_worker.testing.registry.slicing_function import SlicingFunction
 
 
-class Clause:
-    pass
+class Clause(ABC):
+    column: str
+
+    @abstractmethod
+    def mask(self, df: pd.DataFrame) -> pd.Series:
+        ...
+
+    def __repr__(self) -> str:
+        return f"<Clause {str(self)}>"
 
 
 class ComparisonClause(Clause):
-    _operator: str
+    _operator: Callable
```

Comment:
for the sake of completeness can you provide a full signature of this callable (including argument types).